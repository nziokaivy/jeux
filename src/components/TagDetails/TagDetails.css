* {
    box-sizing: border-box;
  }
  .main-game-details {
      padding: 20px;
  }
  /* Create two equal columns that floats next to each other */
  .column-image{
    float: left;
    width: 40%;
    padding: 10px;
  }

  .column-details{
    float: left;
    width: 60%;
    padding: 10px;
  }

  .column-details p {
    font-size: 14px;
  }
  
  /* Clear floats after the columns */
  .row:after {
    content: "";
    display: table;
    clear: both;
  }
  
  .image {
    height: 200px;
    width: 450px;
  }