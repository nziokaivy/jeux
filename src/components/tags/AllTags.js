import React, { Component } from "react";
import { Link } from "react-router-dom";
import "../../assets/css/AllTags.css";
import { fetchAllTags } from '../../redux/actions/Tags/allTagsAction';
import { connect } from "react-redux";
import Header from '../../components/elements/Header/Header';
import HeroImage from '../../components/elements/HeroImage/HeroImage';

export class AllTags extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  async componentDidMount() {
    const { fetchAllTags } = this.props;
    await fetchAllTags();
  }

  render() {
    const { tags } = this.props;

    return (
      <>
      <Header />
      <HeroImage />
      <div>
          <h1 className="allGames-title"> All Tags</h1>
          <hr className='legend' />
        <div className="allGames">
          <div className="flex-grid">
            {tags &&
              tags.results.map(tags => (
                <div key={tags.id} className="col">
                  <img
                    className="allGamesDisplay"
                    src={tags.image_background}
                  />
                  <div className="overlay">
                    <Link to={"/tag/" + tags.id} className="text">
                       {tags.name}
                    </Link>
                  </div>
                </div>
              ))}
          </div>
        </div>
      </div>
      </>
    );
  }
}

export const mapStateToProps = state => ({
  tags: state.allTags.tags,
  tagsError: state.allTags.tagsError
});

export default connect(mapStateToProps, { fetchAllTags })(AllTags);
