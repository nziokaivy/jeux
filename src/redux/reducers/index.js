import { combineReducers } from 'redux';
import allGamesReducer from './allGamesReducer.js';
import gameDetailsReducer from './gameDetailsReducer.js';
import allCreatorsReducer from './allCreatorsReducer.js';
import creatorDetailsReducer from './creatorDetailsReducer';
import allTagsReducer from './allTagsReducer';
import tagDetailsReducer from './tagDetailsReducer';
import allStoresReducer from './allStoresReducer';
import storeDetailsReducer from './storeDetailsReducer';

export default combineReducers({
    allGames: allGamesReducer,
    gameDetails: gameDetailsReducer,
    allCreators: allCreatorsReducer,
    creatorDetails: creatorDetailsReducer,
    allTags: allTagsReducer,
    tagDetails: tagDetailsReducer,
    allStores: allStoresReducer,
    storeDetails:storeDetailsReducer,
});