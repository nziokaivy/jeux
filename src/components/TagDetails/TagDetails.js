import React, { Component } from "react";
import { connect } from "react-redux";
import "./TagDetails.css";
import { fetchTagDetails } from '../../redux/actions/tagDetailsAction';

export class TagDetails extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  componentDidMount() {
    const { id } = this.props.match.params;
    this.props.fetchTagDetails(id);
    let tagInfo;
    if (this.props.fetchTagDetails) {
      tagInfo = this.props.fetchTagDetails;
    }
  }

  render() {
    const { tagDetails, tagDetailsError } = this.props;
    console.log('tagDetails', tagDetails);
    
    return (
        <div className="main-game-details">
        <h1 className="allGames-title"> Tag Details</h1>
        <hr className='legend' />
      <div className="gameDetails">
        {tagDetails && (
          <div className="row">
            <div className="column-image">
              <img
                className="image"
                src={tagDetails.image_background}
              />
            </div>
            <div className="column-details">
              <p><strong>Name:</strong> {tagDetails.name}</p>
              <p><strong>Games Count:</strong> {tagDetails.games_count} games</p>
            </div>
          </div>
        )}
      </div>
      </div>
    );
  }
}
const mapStatetoProps = state => ({
  tagDetails: state.tagDetails.tagsDetails,
  tagDetailsError: state.tagDetails.Error
});
export default connect(mapStatetoProps, { fetchTagDetails })(TagDetails);