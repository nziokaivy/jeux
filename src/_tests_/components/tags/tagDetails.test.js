import React from 'react';
import { shallow } from 'enzyme';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import  { TagDetails, mapStateToProps } from '../../../components/tags/TagDetails';
import rootReducer from '../../../redux/reducers/index';
import mockData from '../../../_mocks_/fileMockData';

const middlewares = [thunk];
let store;

const testStore = (state) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
  return createStoreWithMiddleware(rootReducer, state);
};

const setUp = (initialState = {}) => {
  store = testStore(initialState);
  const wrapper = shallow(
    <TagDetails {...mockData.gameDetailsProps} store={store} />,
  );
  return wrapper;
};

describe('All Creator Details Test Suite', () => {
//   it('Should Pass New props Successfully', () => {
//     const component = setUp(mockData.tagDetailsMainState);
//     //component.setProps({games: mockData.allGamesProps})
//   });

it('Should return initial data', () => {
    const stores = testStore(mockData.tagDetailsMainState);
    expect(shallow(
      <TagDetails  store={stores} />,
    )).toMatchSnapshot();
  });

  it('Should return the initial State', () => {
    const initialState = {
        tagDetails: {
            tagsDetails: null,
            tagsDetailsError: null
           }
    };
    expect(mapStateToProps(initialState).tagDetails).toEqual(null);
  });
});
