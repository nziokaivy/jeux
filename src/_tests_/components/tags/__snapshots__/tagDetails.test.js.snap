// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Creator Details Test Suite Should return initial data 1`] = `ShallowWrapper {}`;
